hydra-core/app/controllers/concerns/hydra/controller/download_behavior.rb

```diff
@@ -120,12 +120,18 @@ def prepare_file_headers
 
       private
 
+      # Previous implementation looped and used response.stream.write, but we
+      # found that had a problem with delay in HTTP response headers being
+      # sent, which this implementation fixed while still streaming.
+      # https://github.com/samvera/hydra-head/pull/421
       def stream_body(iostream)
-        iostream.each do |in_buff|
-          response.stream.write in_buff
+        # see https://github.com/rails/rails/issues/18714#issuecomment-96204444
+        unless response.headers["Last-Modified"] || response.headers["ETag"]
+          Rails.logger.warn("Response may be buffered instead of streaming, best to set a Last-Modified or ETag header")
         end
-      ensure
-        response.stream.close
+
+        render nothing: true
```

Comment:
I wonder if this should be render :head  (see http://guides.rubyonrails.org/layouts_and_rendering.html#using-head-to-build-header-only-responses) It would be more explicit you are putting out headers